src/main/java/beam/physsim/cch/RoutingFrameworkWrapper.scala

```diff
@@ -0,0 +1,207 @@
+                                        | -d binary
+                                        | -o ${graphPathInContainer.toString.replace(".gr.bin", "")}
+                                        | -scc -a way_id capacity coordinate free_flow_speed lat_lng length num_lanes travel_time vertex_id
+      """.stripMargin.replace("\n", ""))
+
+    convertGraphOutput.lineStream.foreach(logger.info(_))
+
+    graphReader.read(graphPathInTempDir.toFile)
+  }
+
+  override def generateOd(): Unit = {
+    val createODPairsOutput = Process(s"""
+                                         |docker run --rm
+                                         | -v $tempDir:/work
+                                         | $toolDockerImage
+                                         | $createODPairsLauncher
+                                         | -g $graphPathInContainer
+                                         | -n 1000
+                                         | -o ${odPairsFileInContainer(0, 0)} -d 10 15 20 25 30 -geom
+      """.stripMargin.replace("\n", ""))
+
+    createODPairsOutput.lineStream.foreach(logger.info(_))
+  }
+
+  def generateOd(iteration: Int, hour: Int, ods: Stream[OD]): Unit = {
+    itHourRelatedPath(tempDirPath, iteration, hour, "").toFile.mkdirs()
+    val odPairsFile = odPairsFileInTempDir(iteration, hour).toFile
+
+    val writer = new BufferedWriter(new FileWriter(odPairsFile))
+    writer.write("origin,destination")
+    writer.newLine()
+
+    ods.foreach {
+      case OD(first, second) =>
+        writer.write(s"$first,$second")
+        writer.newLine()
+    }
+
+    writer.close()
+  }
+
+  override def assignTrafficAndFetchWay2TravelTime(iteration: Int, hour: Int): Map[Long, Double] = {
+    val flowPath = itHourRelatedPath("/work", iteration, hour, "flow")
+    val distPath = itHourRelatedPath("/work", iteration, hour, "dist")
+    val statPath = itHourRelatedPath("/work", iteration, hour, "stat")
+
+    val query = s"""
+                     |docker run --rm
+                     | --memory-swap -1
+                     | -v $tempDir:/work
+                     | $toolDockerImage
+                     | $assignTrafficLauncher
+                     | -g $graphPathInContainer
+                     | -d ${odPairsFileInContainer(iteration, hour)}
+                     | -p 1 
+                     | -n 0 
+                     | -o random
+                     | -i
+                     | ${if (verboseLoggingEnabled) "-v" else ""}
+                     | -flow $flowPath
+                     | -dist $distPath
+                     | -stat $statPath
+      """.stripMargin.replace("\n", "")
+
+    if (verboseLoggingEnabled)
+      logger.info("Docker command for assigning traffic: {}", query)
+
+    val assignTrafficOutput = Process(query)
+
+    assignTrafficOutput.lineStream.foreach(logger.info(_))
+
+    var curIter = -1
+    val wayId2TravelTime = new mutable.HashMap[Long, Double]()
+    Source
```

Comment:
Source will remain open. Please, make sure you close it

Reply:
sure

Reply:
fixed